### PR TITLE
display comments on an article

### PR DESCRIPTION
**Pull request by Holquaria**

### CODE REVIEW

File: src/utils/api.js [hunk 1]
```diff
@@ -19,3 +19,11 @@ export const getArticle = (setArticle, article_id) => {
     }).catch((err) => {
     })
 }
+
+export const getComments = (setComments, article_id) => {
+    return api.get(`/articles/${article_id}/comments`)
+    .then(({data}) => {
+        setComments(data.comments)
+    }).catch((err) => {
```
Comment (popatre) [inline src/utils/api.js]: You dont want the catch here - you need it at the end of the chain, which continues in the app. 
At the moment, you've got two catches.

Comment (popatre) [inline src/utils/api.js]: Please handle the setting of comments in the component. 
A function should only have on responsibility. 
The name of the function is `getComments` - thats all it should do. 

You can then setComments in the comments component. It also make it more reusable, as you might not always want to use the data to set something, you might want to compare it with something, or some other kind of data manipulation.

File: src/components/Comments.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    const [comments, setComments] = useState([])
+    const [loading, setLoading] = useState(false)
+
+    const { article_id } = useParams()
+
+    useEffect(() => {
+        setLoading(true)
+       getComments(setComments, article_id)
+       .then(() => {
+        setLoading(false)
```
Comment (popatre) [inline src/components/Comments.jsx:16]: handle the data setting here

**Review by popatre — CHANGES_REQUESTED** [review]
Could you refactor so that the api data function only has one responsibility. `getting the data` - then you set the data in the component

**Review by popatre — CHANGES_REQUESTED** [review]
Also, could you add something to deal with an article with no comments? Maybe a little message saying no comments to display

**Review by B7Bprog — APPROVED** [review]
Hey Jamie, this is looking good, the path is correct and you give a visual feedback of the page loading. Please merge.

**Pull request merged**